src/Stratis.SmartContracts.Executor.Reflection/InternalTransactionExecutor.cs

```diff
@@ -40,9 +41,59 @@ public sealed class InternalTransactionExecutor : IInternalTransactionExecutor
         }
 
         ///<inheritdoc />
-        public ICreateResult Create<T>(ISmartContractState smartContractState, object[] parameters, ulong amountToTransfer)
+        public ICreateResult Create<T>(ISmartContractState smartContractState, ulong amountToTransfer, CreateContract creationDetails)
         {
-            throw new System.NotImplementedException();
+            // TODO: Expend any neccessary costs.
+
+            ulong gasBudget = creationDetails.GasBudget == 0 ? DefaultGasLimit : creationDetails.GasBudget;
+
+            // Ensure we have enough gas left to be able to fund the new GasMeter.
+            if (smartContractState.GasMeter.GasAvailable < gasBudget)
+                throw new InsufficientGasException();
+
+            var nestedGasMeter = new GasMeter((Gas)gasBudget);
+
+            // Check balance.
+            var balance = smartContractState.GetBalance();
+            if (balance < amountToTransfer)
+            {
+                this.logger.LogTrace("(-)[INSUFFICIENT_BALANCE]:{0}={1}", nameof(balance), balance);
+                throw new InsufficientBalanceException();
+            }
+
+            // Build objects for VM
+            byte[] contractCode = this.contractStateRepository.GetCode(smartContractState.Message.ContractAddress.ToUint160(this.network)); // TODO: Fix this when calling from constructor.
+
```

Comment:
We already have an instance of the Assembly with this Type on it. Rather than retrieving the code and creating another one, can we reuse the existing Assembly?

Reply:
The instance of the assembly we have doesn't necessarily have the correct methods injected with gas spending code. I think overhauling assembly caching should be a separate issue, believe we have one for it #1755

Reply:
Sounds good 👍